spec/features/user_registration_spec.rb

```diff
@@ -0,0 +1,46 @@
+# /spec/features/user_registration_spec.rb
+require 'rails_helper'
+
+RSpec.feature 'New User Registration' do
+  
+  scenario 'A new user registers with valid fields' do
+      visit '/'
+      click_link "Sign Up"
+
+      fill_in "Username", :with => "testuser"
+      fill_in "Password", :with => "password1"
+      fill_in "Email", :with => "test@user.com"
+      fill_in "Age", :wtih => "20"
+      fill_in "Pronoun", :with => "He"
+      fill_in "Zip", :with => "122011"
+      fill_in "City", :with => "Delhi"
+      fill_in "Street Address", :with => "5 Street West"
+      fill_in "Biodata", :with => "All you need to know about me."
+
+      #TODO(soon): Add spec tests for Avatar upload
+
+      click_button "Sign Up"
+      expect(page).to have_text("Welcome testuser!")
+  end
+
+  scenario 'A new user registers with invalid fields' do
+      visit '/'
+      click_link "Sign Up"
+
+      fill_in "Username", :with => "testuser"
```

Comment:
I'm sort of tempted to extract the fields into a given block at the top, like: `given(:user) { build(:user) }` (with a `spec/factories/user.rb` that provides default values for all the fields), and then filling all the fields in with `user.field`. Then it'd be clearer by looking at it what's different between the valid and invalid scenario. But it's optional. Maybe add a TODO.